sdk/python/lib/test/test_resource.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2016-2021, Pulumi Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+import asyncio
+import pytest
+
+from pulumi.runtime import settings, mocks
+from pulumi.runtime.proto import resource_pb2
+import pulumi
+
+
+@pulumi.runtime.test
+def test_depends_on_accepts_outputs(dep_tracker):
```

Comment:
Should we test more permutations of acceptable values for `depends_on` similar to the test for [Node.js](https://github.com/pulumi/pulumi/pull/2198/files#diff-2af4e445f769dffd2cac643d6ec6b50d91fe04731c19eae1267a7fbfefeb4b58)?

https://github.com/pulumi/pulumi/blob/30311405ca1a48f3b3d704981eb5c87e422c2ca9/sdk/nodejs/tests/runtime/langhost/cases/008.ten_depends_on_resources/index.js#L13-L38

Reply:
Didn't see that test, that's pretty helpful, looking.

Reply:
This is great, this discovered a gap wrt supporting "non-listy" versions of the API, they are certainly convenient, will implement those.